nilearn/glm/first_level/first_level.py

```diff
@@ -568,6 +568,11 @@ def compute_contrast(self, contrast_def, stat_type=None,
             raise ValueError('contrast_def must be an array or str or list of'
                              ' (array or str)')
 
+        n_runs = len(self.labels_)
```

Comment:
This is unrelated to the change of this PR, but it seems like the length of `con_vals` is never checked properly. It will work if it is either 1 or `n_runs` but anything else will give an incorrect warning saying that only one contrast was given and will build a larger list than needed which will fail afterwards.

Reply:
Nice catch, I will add a validation of that variable.

Reply:
This is great.
Can you add tests to ensure that the warning/error are raised indeed ?
Thx !